content/articles/compile-your-node.js-application-into-a-.exe-file/index.md

```diff
@@ -0,0 +1,556 @@
+- [References](#References "References")
+
+### Key takeaways
+
+At the end of this tutorial, you would have learned the following:
+
+- What are `.exe` files and their importance.
+- Packages that we can use to compile your Node.js application into `.exe` files.
+- Creating a simple Node.js application.
+- Compiling Node.js files into an executable file for various operating systems.
+- Running the executable file in an operating system.
+
+### Pre-requisites
+
+For one to quickly follow up with this tutorial, you require a minimum of the following requirements:
+
+- Basic web development skills.
+- Some javascript coding skills.
+- Node.js installed in your machine. If not, download it from their [website](https://nodejs.org/en/download/).
+- An IDE. I highly recommend Visual Studio Code which is free and highly customizable. You can download it from [here](https://code.visualstudio.com/download).
+- A stable internet connection.
+
+If you are all set on the above requirements, let's look at some packages to accomplish our goal.
+
+### What are Executable (.exe) files
+
+Executable files (.exe files): These are files containing a set of encoded instructions, which are executed sequentially once the user clicks on the file or runs it.
+
+`.exe` is a Windows extension referring to an executable file. In Linux or Unix-based operating systems, files ending with the `.bin`, `.elf` or `[none]` extensions are executables. In macOS, they lack extensions.
+
+### Pros of executable files
+
+Let us see why the users prefer executable files in the first place. Some advantages of the executable files include:
+
+- **Quick no-code execution:** It only requires one click to run the file, and that's it. Commands are not required to be run on the terminal to execute the program.
+
+- **Prevent unwanted code change:** It prevents unwanted errors generated due to unplanned and accidental modification of source code.
+
+- **Supports distribution:** Being hardware agnostic, the user doesn't have to install any packages or dependencies. This allows one to run multiple instances of the code on any machine without further requirements.
+
+### Packages used
+
+Two of the most commonly used packages used to compile javascript files into executables are:
+
+- [nexe](https://www.npmjs.com/package/nexe "npmjs.com nexe/nexe documentation"): It is a simple command-line utility that compiles your Node.js application into a single executable file. By default, it converts it into a Windows executable. You can quickly jump over to its section by clicking [here](#nexe "nexe").
+- [pkg](https://www.npmjs.com/package/pkg): It is a node package that can convert your node app into several executable files for various operating systems (all at once) or of an individual operating system. You can quickly jump over to its section by clicking [here](#pkg "pkg").
+
+We shall now get a simple node.js application that is running and use it as our application.
```

Comment:
Node.js. Use the proper syntax when referring to Node.js throughout the article.

Reply:
Ok.

Reply:
Done!